src/model/bpmn/internal/shape/ShapeBpmnElement.ts

```diff
@@ -21,6 +21,9 @@ import { ShapeBpmnElementKind, ShapeBpmnEventBasedGatewayKind } from './kinds';
  * @internal
  */
 export default class ShapeBpmnElement {
+  incomingIds?: string[] = [];
+  outgoingIds?: string[] = [];
```

Comment:
These properties are public as they are eventually populated after object instantiation in _ProcessConverter_.

Use an empty array for the default value to not have to deal with _undefined_ in the client code.